src/storage/blockchain/onchain-issuer-adapter/non-merklized/version/v0.0.1/onchain-non-merklized-issuer-adapter.ts

```diff
@@ -0,0 +1,309 @@
+import { Claim, DID, Id } from '@iden3/js-iden3-core';
+import { NonMerklizedIssuerBase__factory } from '@iden3/onchain-non-merklized-issuer-base-abi';
+import { NonMerklizedIssuerBase } from '@iden3/onchain-non-merklized-issuer-base-abi';
+import { INonMerklizedIssuer } from '@iden3/onchain-non-merklized-issuer-base-abi';
+import {
+  CredentialStatusType,
+  DisplayMethod,
+  DisplayMethodType,
+  Iden3SparseMerkleTreeProof,
+  W3CCredential
+} from '../../../../../../verifiable';
+import { Merklizer, Path } from '@iden3/js-jsonld-merklization';
+import { XSDNS } from '../../../../../../circuits';
+import { Hash, Proof } from '@iden3/js-merkletree';
+import { JsonDocumentObject } from '../../../../../../iden3comm';
+import { ethers } from 'ethers';
+import { getDateFromUnixTimestamp } from '@iden3/js-iden3-core';
+import { Options } from '@iden3/js-jsonld-merklization';
+
+enum NonMerklizedIssuerInterfaces {
+  InterfaceDetection = '0x01ffc9a7',
+  InterfaceNonMerklizedIssuer = '0x58874949',
+  InterfaceGetCredential = '0x5d1ca631'
+}
+
+enum ValueHashes {
+  BooleanTrue = '18586133768512220936620570745912940619677854269274689475585506675881198879027',
+  BooleanFalse = '19014214495641488759237505126948346942972912379615652741039992445865937985820'
+}
+
+/**
+ * `OnchainNonMerklizedIssuerAdapter` provides functionality to interact with a non-merklized on-chain credential issuer.
+ * This adapter enables interface detection, credential retrieval, and conversion to the W3C Verifiable Credential format.
+ *
+ * @public
+ * @beta
+ * @class OnchainNonMerklizedIssuerAdapter
+ */
+export class OnchainNonMerklizedIssuerAdapter {
+  private readonly _contract: NonMerklizedIssuerBase;
+  private readonly _contractAddress: string;
+  private readonly _chainId: number;
+
+  private readonly _issuerDid: DID;
+
+  private readonly _merklizationOptions?: Options;
+
+  /**
+   * Initializes an instance of `OnchainNonMerklizedIssuerAdapter`.
+   *
+   * @param address The contract address of the non-merklized issuer.
+   * @param rpcUrl The URL of the blockchain RPC provider.
+   * @param chainId The chain ID of the blockchain network.
+   * @param issuerDid The decentralized identifier (DID) of the issuer.
+   * @param merklizationOptions Optional settings for merklization.
+   */
+  constructor(
+    address: string,
+    options: {
+      rpcUrl: string;
+      chainId: number;
+      issuerDid: DID;
+      merklizationOptions?: Options;
+    }
+  ) {
+    const rpcProvider = new ethers.JsonRpcProvider(options.rpcUrl);
```

Comment:
for now I don't see how it would be possible to use 'did resolver' approach on client with this adapter.
We need to think and implement the fetch of onchain credentials through did resolver too.